src/EditorFeatures/Text/Extensions.SnapshotSourceText.cs

```diff
@@ -334,25 +338,19 @@ public override IReadOnlyList<TextChangeRange> GetChangeRanges(SourceText oldTex
                     }
                 }
 
-                var oldSnapshot = oldText.FindCorrespondingEditorTextSnapshot();
-                var newSnapshot = this.FindCorrespondingEditorTextSnapshot();
+                var oldSnapshot = oldText.FindCorrespondingEditorTextImage();
+                var newSnapshot = this.FindCorrespondingEditorTextImage();
                 return GetChangeRanges(oldSnapshot, oldText.Length, newSnapshot);
             }
 
-            private IReadOnlyList<TextChangeRange> GetChangeRanges(ITextSnapshot oldSnapshot, int oldTextLength, ITextSnapshot newSnapshot)
+            private IReadOnlyList<TextChangeRange> GetChangeRanges(ITextImage oldSnapshot, int oldTextLength, ITextImage newSnapshot)
             {
                 if (oldSnapshot == null ||
-                    newSnapshot == null ||
-                    oldSnapshot.TextBuffer != newSnapshot.TextBuffer)
```

Comment:
ITextImage doesn't point back to buffer, so can't do this directly.  We could try to map back to the editor snapshots, and if that succeeded, we could check the buffers.  What do people think?